### PR TITLE
fix: dynamic cert runnable must NOT be leader election aware

### PR DESCRIPTION
**Pull request by erikgb**
In https://github.com/cert-manager/approver-policy/pull/298 we unintentionally made the dynamic cert runnable leader election aware - which is the default for runnables in controller-runtime.

Since all replicas serve webhook endpoints, the runnable providing webhook certificates must be allowed to run without leader election.

This was probably not caught by our tests since we run only a single replica. And then this is not an issue.

### CODE REVIEW

File: pkg/internal/cmd/cmd.go [hunk 1]
```diff
@@ -143,3 +141,19 @@ func NewCommand(ctx context.Context) *cobra.Command {
 
 	return cmd
 }
+
+type dynamicCertRunnable struct {
```
Comment (inteon) [inline pkg/internal/cmd/cmd.go:145]: Could we make this something generic? So we can re-use it in cert-manager (I think I made the same mistake there).

Comment (erikgb) [reply pkg/internal/cmd/cmd.go:145]: I am not sure if that's possible, but I am not familiar with Golang generics. But I know that generics is not supported for methods, only functions.

Comment (erikgb) [reply pkg/internal/cmd/cmd.go:145]: But feel free to open another PR to fix this. My main concern is to fix the bug.

Comment (inteon) [reply pkg/internal/cmd/cmd.go:145]: I meant making it generic by letting it accept a function as argument (not using actual generics).

**Review by inteon — COMMENTED** [review]
/approve
/lgtm

**Comment by inteon** [comment]
Thanks for fixing the bug @erikgb

File: pkg/internal/cmd/cmd.go (hunk 1 above)
Comment (erikgb) [reply pkg/internal/cmd/cmd.go:145]: Ahhh, sorry for being a bit slow. Can we fix this by making the `DynamicSource` implement these interfaces? That will make it easier to use across projects.

**Pull request merged**